### PR TITLE
feat<table model>: add tablemaster with paddleocr to detect and recognize table

### PR DESCRIPTION
**Pull request by papayalove**
## Motivation

To enhance the ability of table detection and recognition, we adopt a new table structure model——Tablemaster along with paddleocr to deal with tables. TableMaster is way faster than StructEqTable.

## Modification

We add an option for table convertion. Unlike the StructEqTable, the new table model will ouput result in html codes.

## Use cases (Optional)

Change the value of "model" in magic-pdf.json to switch to TableMaster.
```json
{
  // other config
  "models-dir": "D:/models",
  "table-config": {
        "model": "TableMaster", // Another option of this value is 'struct_eqtable'
        "is_table_recog_enable": false, // Table recognition is disabled by default, modify this value to enable it
        "max_time": 400
    }
}
```

## Checklist

**Before PR**:

- [ ] Pre-commit or other linting tools are used to fix the potential lint issues.
- [ ] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: magic_pdf/model/pdf_extract_kit.py [hunk 1]
```diff
@@ -141,10 +151,14 @@ def __init__(self, ocr: bool = False, show_log: bool = False, **kwargs):
         if self.apply_ocr:
             self.ocr_model = ModifiedPaddleOCR(show_log=show_log)
 
-        # init structeqtable
+        # init table model
         if self.apply_table:
-            self.table_model = table_model_init(str(os.path.join(models_dir, self.configs["weights"]["table"])),
-                                                max_time = self.table_max_time, _device_=self.device)
+            if self.table_model_type == STRUCT_EQTABLE:
```
Comment (drunkpig) [inline magic_pdf/model/pdf_extract_kit.py]: 这里可否去掉条件分支改为
```python
table_model_dir = self.configs["weights"][self.table_model_type]

```

File: magic_pdf/model/pdf_extract_kit.py [hunk 2]
```diff
@@ -278,16 +292,28 @@ def crop_img(input_res, input_pil_img, crop_paste_x=0, crop_paste_y=0):
                 new_image, _ = crop_img(res, pil_img)
                 single_table_start_time = time.time()
                 logger.info("------------------table recognition processing begins-----------------")
+                latex_code = None
+                html_code = None
                 with torch.no_grad():
-                    latex_code = self.table_model.image2latex(new_image)[0]
+                    if self.table_model_type == STRUCT_EQTABLE:
```
Comment (drunkpig) [inline magic_pdf/model/pdf_extract_kit.py:295]: 同上，去掉条件分支，代码可以少维护点

**Review by drunkpig — CHANGES_REQUESTED** [review]

File: magic_pdf/model/pdf_extract_kit.py (hunk 2 above)
Comment (papayalove) [reply magic_pdf/model/pdf_extract_kit.py:295]: 这里两种模型产的结果格式不一样，latex和html，还是得判断一下

File: magic_pdf/model/pdf_extract_kit.py (hunk 1 above)
Comment (papayalove) [reply magic_pdf/model/pdf_extract_kit.py]: done

**Review by drunkpig — APPROVED** [review]

**Pull request merged**